code/game/machinery/doors/door.dm

```diff
@@ -144,8 +144,11 @@
 			bound_height = width * world.icon_size
 
 /obj/machinery/door/CanPass(atom/movable/mover, turf/target, height=0)
-	if(istype(mover) && mover.checkpass(PASSGLASS))
-		return !opacity
+	if(istype(mover))
+		if(mover.checkpass(PASSDOOR) && !locked)
+			return TRUE
```

Comment:
```suggestion
		if(mover.checkpass(PASSDOOR))
			return !locked // Disallow if door is bolted
```

Reply:
This did change behaviour. If they have `PASSDOOR` and `PASSGLASS` then it wouldn't matter if the door was glass or not if it was closed. The logic previously would allow you to pass a glass closed door in that case.

Reply:
Ah shoot you are right, these changes will need to be reverted @Contrabang

Reply:
Just undo your last commit and force push